meilisearch-auth/src/key.rs

```diff
@@ -137,3 +132,30 @@ fn generate_id() -> [u8; KEY_ID_LENGTH] {
 
     bytes
 }
+
+fn parse_expiration_date(value: &Value) -> Result<Option<DateTime<Utc>>> {
+    match value {
+        Value::String(string) => DateTime::parse_from_rfc3339(string)
+            .map(|d| d.into())
+            .or_else(|_| {
+                NaiveDateTime::parse_from_str(string, "%Y-%m-%dT%H:%M:%S")
+                    .map(|naive| DateTime::from_utc(naive, Utc))
+            })
+            .or_else(|_| {
+                NaiveDateTime::parse_from_str(string, "%Y-%m-%d")
+                    .map(|naive| DateTime::from_utc(naive, Utc))
+            })
```

Comment:
Shouldn't we choose one format and stick to it instead of trying to handle everything? :thinking: